File: Solutions/Sheep and Coyotes.py
import sys
input = sys.stdin.readline


def distance(x1, y1, x2):
    return float(((x1 - x2)**2 + y1**2)**(1/2))


n = int(input())
sheepx, sheepy = [], []
xc = [False]*100001
lg = 0
for _ in range(n):
    x, y = float(input()), float(input())
    if xc[int(x*100)]:
        cur = sheepx.index(x)
        if sheepy[cur] > y:
            sheepy[cur] = y
    else:
        sheepx.append(x)
        sheepy.append(y)
        xc[int(x*100)] = True
        if x*100 > lg:
            lg = x*100
eat = [False]*(len(sheepx))
for i in range(100001):
    if xc[i] or i > lg:
        s = 9999
        ind = []
        for j in range(len(sheepx)):
            x = distance(sheepx[j], sheepy[j], i / 100)
            if x < s:
                ind = [j]
                s = x
            elif x == s:
                ind += [j]
        for j in ind:
            eat[j] = True
for i in range(len(sheepx)):
    if eat[i]:
        print('The sheep at (' + str("%.2f" % sheepx[i]) + ', ' + str("%.2f" % sheepy[i]) + ') might be eaten.')

File: Solutions/Sheep and Coyotes.cpp
//
// Created by Leon on 2019-02-12.
//
#include <bits/stdc++.h>

using namespace std;

int n;
double sheep [100][2];
bool eat [100];
int main() {
    cin >> n;
    for(int i = 0; i < n; i++){
        cin >> sheep[i][0] >> sheep[i][1];
    }
    for(float i = 0.00; i < 1000.01; i += 0.01){
        int ind = 0;
        double s = pow((sheep[0][0] - i), 2) + pow(sheep[0][1], 2);
        for(int j = 1; j < n; j++){
            double x = pow((sheep[j][0] - i), 2) + pow(sheep[j][1], 2);
            if(x < s){
                ind = j;
                s = x;
            }
        }
        eat[ind] = true;
    }
    for(int i = 0; i < n; i++){
        if(eat[i]){
            printf("The sheep at (");
            printf("%.2f", sheep[i][0]);
            printf(", ");
            printf("%.2f", sheep[i][1]);
            printf(") might be eaten. \n");
        }
    }
    return 0;
}
